File: scripts/generate-__init__.py.py
#!/usr/bin/env python
"""Generate ``src/*/external/*/__init__.py``."""
import os
import sys

name = os.path.dirname(sys.argv[1]).split("external/")[-1].replace("/", ".")
char = "="
head = '"""Fake '
with open(sys.argv[2]) as f:
    print(
        f.read()
        .replace("__init__", name)
        .replace(char * len(head + "__init__"), char * len(head + name)),
        end="",
    )

File: scripts/generate-__init__.py.pl
#!/usr/bin/env -S perl -p
BEGIN {
  $name = shift;
  $name =~ s=.*/external/(.+?)/__init__\.py=$1=;
  $name =~ s=/=.=g;
}
s/shtab/$name/g;
$mark = "=" x length $name;
s/=====/$mark/;
